src/Driver/Mysqli/Result.php

```diff
@@ -40,11 +45,17 @@ final class Result implements ResultInterface
     /**
      * @internal The result can be only instantiated by its driver connection or statement.
      *
+     * @param Statement|null $statementReference This reference is to not trigger destruction
+     * of the Statement object too early.
```

Comment:
Let's move this remark to the property's doc block.

Reply:
Tidied up the comment a bit  and move everything to the property.